### PR TITLE
Break hung rbd list if cluster in error state

### PR DESCRIPTION
**Pull request by shtripat**
If cluster is in error state, rbd commands get hung. Added a logic
to break the hung process and continue with other sync operations.

Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
@Tendrl/tendrl-core plz review

File: tendrl/ceph_integration/ceph.py [hunk 1]
```diff
@@ -380,9 +380,14 @@ def rbd_command(command_args, pool_name=None):
 
     p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                          stdin=open(os.devnull, "r"))
+    if p.poll() is None: # Force kill if process is still alive
```
Comment (r0h4n) [inline tendrl/ceph_integration/ceph.py:384]: How much seconds before you consider the rbd cmd as not responding due to cluster in ERROR state?

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: tendrl/ceph_integration/ceph.py (hunk 1 above)
Comment (shtripat) [reply tendrl/ceph_integration/ceph.py:384]: Our sync runs every 3 seconds and rbd commands are pretty faster so kept the wait as 1 second here, and by that time if its not responding I terminate.
To be safer if you want we can make this at least 2 seconds. What you feel?

Comment (r0h4n) [reply tendrl/ceph_integration/ceph.py:384]: - time.sleep is bad when using gevent use gevent.sleep(2)

Comment (shtripat) [reply tendrl/ceph_integration/ceph.py:384]: Done. Will change.

**Review by r0h4n — APPROVED** [review]

**Pull request merged**